internal/pkg/manifest/workload.go

```diff
@@ -289,3 +302,30 @@ func UnmarshalWorkload(in []byte) (interface{}, error) {
 		return nil, &ErrInvalidWorkloadType{Type: typeVal}
 	}
 }
+
+func requiresBuild(image Image) (bool, error) {
+	hasBuild, hasURL := image.Build.isEmpty(), image.Location == nil
+	if hasBuild == hasURL {
+		return false, fmt.Errorf(`either "image.build" or "image.location" needs to be specified in the manifest`)
+	}
+	if image.Location == nil {
+		return true, nil
+	}
+	return false, nil
+}
+
+// DockerfileBuildRequired returns if the container image should be built from local Dockerfile.
+func DockerfileBuildRequired(svc interface{}) (bool, error) {
+	type manifest interface {
+		BuildRequired() (bool, error)
+	}
+	mf, ok := svc.(manifest)
+	if !ok {
+		return false, fmt.Errorf("workload does not have required methods BuildRequired()")
```

Comment:
is there a way we can phrase this without using "workload" since we've taken such pains to not surface it to customers elsewhere?